game-core/src/main/java/games/strategy/engine/framework/startup/ui/LobbyWatcherKeepAliveTask.java

```diff
@@ -45,7 +48,9 @@ public void run() {
           reportReconnected();
         }
       } else {
-        gameId = gamePoster.get();
+        final PostGameResponse postGameResponse = gamePoster.get();
+        gameId = postGameResponse.getGameId();
+        connectivityHandler.accept(postGameResponse.isSuccessfulConnectivityCheck());
```

Comment:
This can pose a major problem. Let's say a user disconnects from a network that has a direct connection to one that no longer does (which is about the only way you'd be able to create and post a game and then later have connectivity fail on a re-posting).

In that scenario the keep alive sender would no-op while disconnected, by design, but when we get a reconnection we would get a new game and then the connectivity check would fail. The connectivity handler would then crash the game that is then in-progress. That game crash would be quite severe, no option to save, the game just closes down on you with an error message.

One could simply not do the connectivity check here, which gets into my point a bit about interface segregation principle. I'm certainly leaning to decoupling the functionality so that we don't have dead code on certain call paths. Otherwise if the two functionalities remain coupled, then we need a different error handler implementation to not crash an already-in-progress game.

Reply:
IF post-game can return a nullable gameId, the state here becomes a bit more complex as we now have potential for a null game id. That updates the test scenarios we need to account for the increased state complexity

Reply:
Another item to consider with this code, we should not proceed to do another keep alive check if the connectivity check fails. The code here instead assumes the implementation of `connectivityHandler` will crash the game so that we do not proceed.

To keep the abstraction intact, we would probably want connectivity handler to be a predicate to return true/false if we should continue processing, but then again this code is already aware of the connectivity check result and probably could just no-op if the connectivity check is false. Messaging that well to a user is potentially a headache though.